NNLOPS in new reweighting scheme

covarell opened the pull request:
NNLOPS for powheg in new reweighting scheme.

Tested OK, but still needs editing resulting LHE files with scripts if the main MINLO-HJ run has a lot of weights (a lot means > 10...)

govoni commented:
ciao Roberto, what does it mean that editing is still needed?

covarell commented:
It means the merging of the resulting LHE files is done using a trick, so that NNLOPS is blind to the hundreds of PDF weights and just sees the scale variations needed to run (you can have a look at the run_nnlops.sh script). This is what I also used to produce current NNLOPS samples.

govoni commented:
is this preventing an automated running of the NNLOPS with the usual gridpack scheme in a wmLHE campaign?

covarell commented:
Obviously. I believe this is a well established fact by now (regardless of the POWHEG version). 

NNLOPS reweighting is now understood to work only on a minlo LHE file containing a few 100k's of events and it is not possible to instruct wmLHE to produce so many events in one shot.

govoni commented:
:D :D

perrozzi commented on bin/Powheg/run_pwg.py:
```diff
@@ -595,33 +600,33 @@ def runGetSource(parstage, xgrid, folderName, powInputName, process, tagName) :
-<weight id='nnlops-0505-7'> combines 'nn-mtmb-0505' with '1007' </weight>
-<weight id='nnlops-0505-8'> combines 'nn-mtmb-0505' with '1008' </weight>
-<weight id='nnlops-0505-9'> combines 'nn-mtmb-0505' with '1009' </weight>
+<weight id='nnlops-11-1'> combines 'nn-mtmb-11' with '1000' </weight> 
```
may I ask what is the reason to shift this?

covarell replied:
No idea. It is like this in the make-pwg-wl.py script and I followed the new convention.
@yuanchao  ?

yuanchao replied:
I don't understand either. Also that I'd prefer all weighting stuff should go to make-pwg-wl.py.

covarell replied:
@yuanchao 
In this script (written by you)
https://github.com/cms-sw/genproductions/blob/powheg_gcc530/bin/Powheg/make_rwl.py
the weights start all from a round number (e.g. 1000, 2000 etc.) because the counter is increased at the end of the loop. While in the past they used to start from a round number + 1. Did you mean to make this change? Otherwise, please modify the script and I will eliminate the shift from NNLOPS weights.

yuanchao commented on bin/Powheg/run_pwg.py:
```diff
@@ -338,7 +338,7 @@ def runGetSource(parstage, xgrid, folderName, powInputName, process, tagName) :
 ### retrieve the powheg source tar ball
-export POWHEGSRC=powhegboxV2rwgt_Mar2017.tar.gz
+export POWHEGSRC=powhegboxV2_rev3407_date20170711.tar.gz
 
```
The powhegbox source package naming would need some discussion? Here 'date' is a bit redundant. Surely 'rwgt' is not needed anymore.

covarell replied:
rwgt is in fact removed. 
The new naming is given automatically by the create_source script (of which I am also not an author)

covarell commented:
I have shifted all weights by 1 in the make_rwl script in the last commit since it seems we all agree on starting from 1001, 2001 etc.

Is there a final agreement already on the PDF sets for 2017 NLO and NNLO samples? If so, we can make standard script for the general use.

perrozzi commented on bin/Powheg/run_pwg.py:
```diff
@@ -679,10 +684,6 @@ def runGetSource(parstage, xgrid, folderName, powInputName, process, tagName) :
 fi
 
-
-sed -i s/SCRAM_ARCH_VERSION_REPLACE/${SCRAM_ARCH}/g runcmsgrid.sh
```
could it be that you started modifying the script before updating to the central repository?
I recently introduced these lines to ensure the back-/forward- compatibility of the gridpacks in production,
unless there is a specific reason to remove them they need to be there :-)

perrozzi commented on bin/Powheg/run_pwg.py:
```diff
@@ -816,14 +817,10 @@ def createTarBall(parstage, folderName, prcName, keepTop, seed, scriptName) :
-fi  
-
-
-sed -i s/SCRAM_ARCH_VERSION_REPLACE/${SCRAM_ARCH}/g runcmsgrid.sh
```
same here

perrozzi commented on bin/Powheg/run_pwg.py:
```diff
@@ -854,12 +851,11 @@ def createTarBall(parstage, folderName, prcName, keepTop, seed, scriptName) :
 if [ $keepTop == '1' ]; then
     echo 'Keeping validation plots.'
-    echo 'Packing...' ${WORKDIR}'/'${folderName}'_'${SCRAM_ARCH}'_'${CMSSW_VERSION}'_'${process}'.tgz'
-    tar zcf ${WORKDIR}'/'${folderName}'_'${SCRAM_ARCH}'_'${CMSSW_VERSION}'_'${process}'.tgz' * --exclude=POWHEG-BOX --exclude=powhegbox*.tar.gz --exclude=*.lhe --exclude=run_*.sh --exclude=*.log --exclude=*temp --exclude=pwgbtlupb-*.dat --exclude=pwgrmupb-*.dat
```
same here

perrozzi commented on bin/Powheg/run_pwg.py:
```diff
@@ -854,12 +851,11 @@ def createTarBall(parstage, folderName, prcName, keepTop, seed, scriptName) :
+    tar zcf ${WORKDIR}'/'${folderName}'_'${process}'.tgz' * --exclude=POWHEG-BOX --exclude=powhegbox*.tar.gz --exclude=*.lhe --exclude=run_*.sh --exclude=*.log --exclude=*temp --exclude=pwgbtlupb-*.dat --exclude=pwgrmupb-*.dat
 else
-
-    echo 'Packing...' ${WORKDIR}'/'${folderName}'_'${SCRAM_ARCH}'_'${CMSSW_VERSION}'_'${process}'.tgz'
```
same here

govoni commented:
@covarell the agreement about pdfs is here:
https://docs.google.com/spreadsheets/d/1YjhruLPvmwqoA1YZPgmU5H9T3ELnYnriHgO9DrJQoxk/edit#gid=0

covarell commented:
Ok, fixed wrong backports (@perrozzi, one of the comments, the one about setting produceWeightsNNLO should not be changed, otherwise weights are calculated on the fly) and updated pdf sets

govoni commented:
thanks. Are the weights ok for any powheg production already?

perrozzi commented on bin/Powheg/make_rwl.py:
```diff
@@ -20,47 +20,51 @@
 for m_rensc in m_factor :
   for m_facsc in m_factor :
-    fout.write("<weight id='"+str(m_idx)+"'> lhapdf=260000 renscfact="+ \
+    fout.write("<weight id='"+str(m_idx)+"'> lhapdf=305800 renscfact="+ \
```
I believe we want to go to NNLO for the central value, there is a mistake in the google spreadsheet, therefore 306000 and similar

covarell replied:
What do you mean? NNLO for NLO samples???

perrozzi replied:
and LO samples too :-)

govoni commented:
@covarell yes, after a long discussion during GEN meetings and pdf4lhc forum meetings.

govoni commented:
@perrozzi can we merge?

perrozzi commented:
the main purpose of the PR is to update stuff for NNLOPS, which is done. PDF are not fully updated yet but can be done in a future PR.
so yes I think we can merge

govoni commented:
ok. Roberto, we assume there will be a dedicated PR for the PDFs, then.
It looks they deserve one :)
